core/src/main/java/org/mule/runtime/core/internal/exception/ErrorTypeLocatorFactory.java

```diff
@@ -85,6 +85,7 @@ public static ErrorTypeLocator createDefaultErrorTypeLocator(ErrorTypeRepository
                                  errorTypeRepository.lookupErrorType(STREAM_MAXIMUM_SIZE_EXCEEDED).get())
             .addExceptionMapping(MuleFatalException.class, errorTypeRepository.getErrorType(FATAL).get())
             .addExceptionMapping(CompositeRoutingException.class, errorTypeRepository.getErrorType(COMPOSITE_ROUTING).get())
+            .addExceptionMapping(LinkageError.class, errorTypeRepository.getErrorType(FATAL).get())
```

Comment:
Why was this necessary? AFAIK that error should be wrapped to a MuleFatalException.